**Native Tests main** fails with many occurrences of:

```
Caused by: com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfaces [interface javax.servlet.http.HttpServletRequest] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
```

Does anybody have an idea?

--- a/extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java
+++ b/extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java
@@ -131,13 +131,15 @@ class VertxWebProcessor {
                 generatedClass.produce(new GeneratedClassBuildItem(true, name, data));
             }
         };
+        Handler<RoutingContext> bodyHandler = recorder.createBodyHandler(httpConfiguration);
+
         for (AnnotatedRouteHandlerBuildItem businessMethod : routeHandlerBusinessMethods) {
             String handlerClass = generateHandler(businessMethod.getBean(), businessMethod.getMethod(), classOutput);
             reflectiveClasses.produce(new ReflectiveClassBuildItem(false, false, handlerClass));
             Handler<RoutingContext> routingHandler = recorder.createHandler(handlerClass);
             for (AnnotationInstance routeAnnotation : businessMethod.getRoutes()) {
                 Route route = annotationProxy.builder(routeAnnotation, Route.class).build(classOutput);
-                Function<Router, io.vertx.ext.web.Route> routeFunction = recorder.createRouteFunction(route, httpConfiguration);
+                Function<Router, io.vertx.ext.web.Route> routeFunction = recorder.createRouteFunction(route, bodyHandler);
                 AnnotationValue typeValue = routeAnnotation.value("type");
                 HandlerType handlerType = HandlerType.NORMAL;
                 if (typeValue != null) {

--- a/extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/VertxWebRecorder.java
+++ b/extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/VertxWebRecorder.java
@@ -36,7 +36,7 @@ public class VertxWebRecorder {
     }
 
     public Function<Router, io.vertx.ext.web.Route> createRouteFunction(Route routeAnnotation,
-            HttpConfiguration httpConfiguration) {
+            Handler<RoutingContext> bodyHandler) {
         return new Function<Router, io.vertx.ext.web.Route>() {
             @Override
             public io.vertx.ext.web.Route apply(Router router) {
@@ -66,22 +66,24 @@ public class VertxWebRecorder {
                         route.consumes(consumes);
                     }
                 }
-
-                BodyHandler bodyHandler = BodyHandler.create();
-                Optional<MemorySize> maxBodySize = httpConfiguration.limits.maxBodySize;
-                if (maxBodySize.isPresent()) {
-                    bodyHandler.setBodyLimit(maxBodySize.get().asLongValue());
-                }
-                final BodyConfig bodyConfig = httpConfiguration.body;
-                bodyHandler.setHandleFileUploads(bodyConfig.handleFileUploads);
-                bodyHandler.setUploadsDirectory(bodyConfig.uploadsDirectory);
-                bodyHandler.setDeleteUploadedFilesOnEnd(bodyConfig.deleteUploadedFilesOnEnd);
-                bodyHandler.setMergeFormAttributes(bodyConfig.mergeFormAttributes);
-                bodyHandler.setPreallocateBodyBuffer(bodyConfig.preallocateBodyBuffer);
-
                 route.handler(bodyHandler);
                 return route;
             }
         };
+    }
+
+    public Handler<RoutingContext> createBodyHandler(HttpConfiguration httpConfiguration) {
+        BodyHandler bodyHandler = BodyHandler.create();
+        Optional<MemorySize> maxBodySize = httpConfiguration.limits.maxBodySize;
+        if (maxBodySize.isPresent()) {
+            bodyHandler.setBodyLimit(maxBodySize.get().asLongValue());
+        }
+        final BodyConfig bodyConfig = httpConfiguration.body;
+        bodyHandler.setHandleFileUploads(bodyConfig.handleFileUploads);
+        bodyHandler.setUploadsDirectory(bodyConfig.uploadsDirectory);
+        bodyHandler.setDeleteUploadedFilesOnEnd(bodyConfig.deleteUploadedFilesOnEnd);
+        bodyHandler.setMergeFormAttributes(bodyConfig.mergeFormAttributes);
+        bodyHandler.setPreallocateBodyBuffer(bodyConfig.preallocateBodyBuffer);
+        return bodyHandler;
     }
 }

